File: 589. N-ary Tree Preorder Traversal/solution.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        out = []
        if root == None:
            return out
        out.append(root.val)
        if root.children != None:
            for c in root.children:
                out += self.preorder(c)
        return out

File: 589. N-ary Tree Preorder Traversal/solution.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func preorder(root *Node) []int {
    var curr []int
    if root == nil {
        return curr
    }
    curr = append(curr, root.Val)
    if len(root.Children) == 0 {
        return curr
    } 
    for ii := 0; ii < len(root.Children); ii++ {
        temp := preorder(root.Children[ii])
        for jj := 0; jj < len(temp); jj++ {
            curr = append(curr, temp[jj])
        }
    }
    return curr
}
